src/cli/cli.cpp

```diff
@@ -708,6 +719,90 @@ void Interpreter::ProcessDiscover(int argc, char *argv[])
     AppendResult(error);
 }
 
+#if OPENTHREAD_ENABLE_DNS_CLIENT
+void Interpreter::ProcessDns(int argc, char *argv[])
+{
+    ThreadError error = kThreadError_None;
+    long port = OT_DNS_DEFAULT_SERVER_PORT;
+    Ip6::MessageInfo messageInfo;
+    otDnsQuery query;
+
+    if (argc && strcmp(argv[0], "resolve") == 0)
+    {
+        VerifyOrExit(!mResolvingInProgress, error = kThreadError_Busy);
+        VerifyOrExit(argc >= 2, error = kThreadError_Parse);
+        VerifyOrExit(strlen(argv[1]) < OT_DNS_MAX_HOSTNAME_LENGTH, error = kThreadError_Parse);
+
+        strcpy(mResolvingHostname, argv[1]);
+
+        memset(&messageInfo, 0, sizeof(messageInfo));
+        messageInfo.mInterfaceId = OT_NETIF_INTERFACE_ID_THREAD;
+
+        if (argc > 2)
+        {
+            SuccessOrExit(error = messageInfo.GetPeerAddr().FromString(argv[2]));
+        }
+        else
+        {
+            // Use Google Public IPv6 DNS Server.
+            SuccessOrExit(error = messageInfo.GetPeerAddr().FromString("2001:4860:4860::8888"));
```

Comment:
Could this default value be put in a `#define`?  Possibly as `OT_DNS_DEFAULT_DNS_SERVER` in dns.h?